src/sentry/static/sentry/app/components/createAlertButton.tsx

```diff
@@ -0,0 +1,269 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {Project, Organization} from 'app/types';
+import {t, tct} from 'app/locale';
+import {IconInfo, IconClose, IconSiren} from 'app/icons';
+import Button from 'app/components/button';
+import EventView from 'app/utils/discover/eventView';
+import Alert from 'app/components/alert';
+import space from 'app/styles/space';
+import {explodeFieldString} from 'app/utils/discover/fields';
+import {
+  errorFieldConfig,
+  transactionFieldConfig,
+} from 'app/views/settings/incidentRules/constants';
+
+/**
+ * Discover query supports more features than alert rules
+ * To create an alert rule from a discover query, some parameters need to be adjusted
+ */
+type IncompatibleQueryProperties = {
+  /**
+   * Must have exactly one project selected and not -1 (all projects)
+   */
+  hasProjectError: boolean;
+  /**
+   * Must have zero or one environments
+   */
+  hasEnvironmentError: boolean;
+  /**
+   * event.type must be error or transaction
+   */
+  hasEventTypeError: boolean;
+  hasYAxisError: boolean;
+};
+
+type AlertProps = {
+  /**
+   * Called when the eventView does not meet the requirements of alert rules
```

Comment:
What do you mean `Called`? This isn't a function right?

Reply:
this comment was a mistake, thanks!